fancy-weather/src/index.js

```diff
@@ -0,0 +1,426 @@
+  setTimeout(setDateAndTime, 1000);
+}
+
+function toggleButtonLanguage() {
+  elementTownInput.placeholder = `${dictionary[currentLanguage].buttons.placeholder}`;
+  elementSearchButton.value = `${dictionary[currentLanguage].buttons.search}`;
+}
+
+function toggleCurrentLanguage() {
+  currentLanguage = elementLanguageButton.value;
+  localStorage.setItem('language', currentLanguage);
+  toggleButtonLanguage();
+  if (data.geocoding.city) {
+    controlWeather();
+  }
+}
+
+function loadSaveLanguage() {
+  if (localStorage.getItem('language') === null) {
+    localStorage.setItem('language', 'en');
+  } else {
+    currentLanguage = localStorage.getItem('language');
+    elementLanguageButton.value = localStorage.getItem('language');
+  }
+  toggleButtonLanguage();
+}
+
+function convertCelsiumFahrenheit(value) {
+  let result;
+  if (data.weather.scale === 'celsium') {
+    result = value * 1.8 + 32;
+  }
+  if (data.weather.scale === 'fahrenheit') {
+    result = (value - 32) / 1.8;
+  }
+  return result;
+}
+
+function setCurrentWeatherScale() {
+  if (currentScale !== data.weather.scale) {
+    data.weather.currentTemp = convertCelsiumFahrenheit(data.weather.currentTemp);
+    data.weather.currentFeelsLike = convertCelsiumFahrenheit(data.weather.currentFeelsLike);
+    data.weather.nextFirstDayTemp = convertCelsiumFahrenheit(data.weather.nextFirstDayTemp);
+    data.weather.nextSecondDayTemp = convertCelsiumFahrenheit(data.weather.nextSecondDayTemp);
+    data.weather.nextThirdDayTemp = convertCelsiumFahrenheit(data.weather.nextThirdDayTemp);
+    if (data.weather.scale === 'celsium') {
+      data.weather.scale = 'fahrenheit';
+    } else {
+      data.weather.scale = 'celsium';
+    }
+    if (elementWeatherLocation.innerHTML !== '') {
+      fillDOMContent();
+    }
+  }
+}
+
+function toggleTemperatureButton() {
+  if (currentScale === 'fahrenheit') {
+    elementFahrenheitButton.classList.remove('transparent');
+    elementCelsiumButton.classList.add('transparent');
+  }
+  if (currentScale === 'celsium') {
+    elementCelsiumButton.classList.remove('transparent');
+    elementFahrenheitButton.classList.add('transparent');
+  }
+  localStorage.setItem('scale', currentScale);
+  setCurrentWeatherScale();
+}
+
+function loadSaveScale() {
+  if (localStorage.getItem('scale') === null) {
+    localStorage.setItem('scale', 'celsium');
+  } else {
+    currentScale = localStorage.getItem('scale');
+  }
+  toggleTemperatureButton();
+}
+
+function getLocationByIP() {
+  return fetch('https://ipinfo.io/json?token=72f4ef7fa30822')
+    .then((response) => response.json())
+    .then((jsonResponse) => {
+      console.log(jsonResponse);
+      coordinatesIP = jsonResponse.loc;
+      return data;
+    });
+}
+
+function getCoordinates(query) {
+  return fetch(`https://api.opencagedata.com/geocode/v1/json?q=${query}&key=7dd8ba7a84bd4d0fa897bf84577d8693&language=${currentLanguage}&no_annotations=1&limit=1`)
```

Comment:
Hi @andreiha 
It's better to have config or const for such value us  ``key=7dd8ba7a84bd4d0fa897bf84577d8693&language``